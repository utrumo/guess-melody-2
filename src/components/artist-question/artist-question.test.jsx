import React from 'react';
import renderer from 'react-test-renderer';
import ArtistQuestion from './artist-question.jsx';
import createAudioNodeMock from '../../mocks/audio-node-mock';

it(`ArtistQuestionScreen is rendered correctly`, () => {
  const mockQuestion = {
    type: `artist`,
    song: {
      artist: `Quincas Moreira`,
      src: `https://es31-server.appspot.com/guess-melody/static/music/Blue_Whale.mp3`
    },
    answers: [
      {
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
        artist: `Quincas Moreira`
      },
      {
        artist: `Density & Time`,
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`
      },
      {
        artist: `Endless Love`,
        picture: `https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg`
      }
    ]
  };
  const step = 2;
  const onUserAnswer = jest.fn();
  const artistQuestionScreen = renderer
    .create(
        <ArtistQuestion
          question={mockQuestion}
          step={step}
          onAnswer={onUserAnswer}
        />,
        createAudioNodeMock
    )
    .toJSON();

  expect(artistQuestionScreen).toMatchSnapshot();
});
