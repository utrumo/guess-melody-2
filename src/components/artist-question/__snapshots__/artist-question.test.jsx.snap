// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen is rendered correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <button
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onClick={[Function]}
        type="button"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Quincas Moreira"
          className="artist__picture"
          src="https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
        />
        Quincas Moreira
      </label>
    </div>
    <div
      className="artist"
    >
      <button
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onClick={[Function]}
        type="button"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Density & Time"
          className="artist__picture"
          src="https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg"
        />
        Density & Time
      </label>
    </div>
    <div
      className="artist"
    >
      <button
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onClick={[Function]}
        type="button"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Endless Love"
          className="artist__picture"
          src="https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
        />
        Endless Love
      </label>
    </div>
  </form>
</section>
`;
